go/kbfs/libkbfs/folder_block_manager.go

```diff
@@ -1379,6 +1379,30 @@ func (fbm *folderBlockManager) doCleanDiskCache(cacheType DiskBlockCacheType) (
 		// archive races (see comment below).
 		ptrs := getUnrefPointersFromMD(rmd, true)
 
+		// Cancel any prefetches for these blocks that might be in
+		// flight, to make sure they don't get put into the cache
+		// after we're done cleaning it.  Ideally we would cancel them
+		// in a particular order (the lowest level ones first, up to
+		// the root), but since we already do one round of
+		// prefetch-canceling as part of applying the MD and updating
+		// the pointers, doing a second round here should be good
+		// enough to catch any weird relationships between the
+		// pointers where one non-yet-canceled prefetch can revive the
+		// prefetch of an already-canceled child block.
+		for _, ptr := range ptrs {
+			fbm.config.BlockOps().Prefetcher().CancelPrefetch(ptr)
+			c, err := fbm.config.BlockOps().Prefetcher().
+				WaitChannelForBlockPrefetch(ctx, ptr)
+			if err != nil {
+				return err
+			}
+			select {
+			case <-c:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
```

Comment:
Why wait one at a time instead of waiting for all of them? 

I don't see why this should improve the probability of something triggering another and getting through.

Reply:
It's harder to code a select across a variable number of channels, and I don't think it would improve performance much, since the prefetcher is single-goroutined anyway.  This way seems both easier to understand and easier to code.

Reply:
Also, performance is not really a concern here anyway, since it's a background goroutine that shouldn't ever block the user.

Reply:
Didn't think about the prefetcher being single threaded, seems reasonable.